### PR TITLE
test: add config generation stability tests

### PR DESCRIPTION
**Pull request by smira**
These tests ensure that config generation for older versions of Talos is stable as we move forward and introduce new features in the machine configuration.

### CODE REVIEW

File: pkg/machinery/config/types/v1alpha1/v1alpha1_stability_test.go [hunk 1]
```diff
@@ -0,0 +1,121 @@
...
+	const generateMode = false
+
+	secretsBundle, err := secrets.LoadBundle("testdata/stability/secrets.yaml")
+	require.NoError(t, err)
+
+	for _, versionContract := range []*config.VersionContract{
+		config.TalosVersion1_3,
+		config.TalosVersion1_4,
+		config.TalosVersion1_5,
+		config.TalosVersion1_6,
```
Comment (utkuozdemir) [inline pkg/machinery/config/types/v1alpha1/v1alpha1_stability_test.go:38]: should we make this fail explicitly if there's a newer version untested? so we don't forget it in the future?

**Review by utkuozdemir — APPROVED** [review]

File: pkg/machinery/config/types/v1alpha1/v1alpha1_stability_test.go (hunk 1 above)
Comment (smira) [reply pkg/machinery/config/types/v1alpha1/v1alpha1_stability_test.go:38]: I can do that, good point

**Review by Unix4ever — APPROVED** [review]

File: pkg/machinery/config/types/v1alpha1/v1alpha1_stability_test.go [hunk 2]
```diff
@@ -0,0 +1,130 @@
...
+	"github.com/siderolabs/talos/pkg/machinery/config/types/v1alpha1"
+	"github.com/siderolabs/talos/pkg/machinery/gendata"
+)
+
+// TestConfigEncodingStability ensures that the encoding of a configuration is stable as we moved forward with the config format.
+func TestConfigEncodingStability(t *testing.T) {
+	t.Parallel()
+
+	// flip this to generate missing configs
+	const generateMode = false
```
Comment (utkuozdemir) [inline pkg/machinery/config/types/v1alpha1/v1alpha1_stability_test.go:32]: cool trick

File: pkg/machinery/config/types/v1alpha1/v1alpha1_stability_test.go (hunk 1 above)
Comment (smira) [reply pkg/machinery/config/types/v1alpha1/v1alpha1_stability_test.go:41]: added a check

**Review by utkuozdemir — APPROVED** [review]

**Pull request merged**